src/function/execute.rs

```diff
@@ -54,6 +54,15 @@ where
             // previous execution as the starting point for the new one.
             if let Some(old_memo) = opt_old_memo {
                 active_query.seed_tracked_struct_ids(&old_memo.revisions.tracked_struct_ids);
+
+                // Copy over all outputs from a previous iteration.
+                // This is necessary to ensure that tracked struct created during the previous iteration
+                // (and are owned by the query) alive even if the query in this iteration no longer creates them.
+                // The query not re-creating the tracked struct doesn't guarantee that there
+                // aren't any other queries depending on it.
+                if old_memo.may_be_provisional() && old_memo.verified_at.load() == revision_now {
+                    active_query.seed_outputs(old_memo.revisions.origin.outputs());
```

Comment:
I am not sure about the test failure but given that the order of inputs and outputs for dependencies is fairly important, it might be more correct to seed the outputs *after* having executed the query such that the seeding does not influence the executions dependency order? Maybe that is causing issues

Reply:
Let me try this. I'd be surprised if it's the case because `input_outputs` isn't used for ID seeding. But who knows, maybe it changes `maybe_changed_after` execution order (although it shouldn't?)

Reply:
Same, but figured it might be worth a try. Seems to not have been the case though

Reply:
I'll keep this regardless because I think it's nice to keep the input/output order as close as possible to the actual execution order.